### DISCUSSION TITLE
Song selection menu feels unresponsive

### DISCUSSION THREAD

**Discussion by Alawapr** [ppy/osu#24916] [Q&A] [answered]
### Type

Performance

### Bug description

When going through maps in the song selection menu in osu!lazer (Watch first 2 videos), it feels extremely unresponsive, it's especially noticeable if you focus on the delay that occurs between when you switch to another map till the background of the map changes and the song starts playing.
This is further emphasized if you compare it to how osu!stable performs while doing the same thing (Watch 3rd and 4th video).

For correctness, I asked a friend to also record and perform the same actions, which resulted in the same thing: osu!lazer feels worse.

This has been happening not only in this version of osu!lazer but in past ones as well from what I can remember.

### Screenshots or videos

# osu!lazer (First is my video, second is Colgate's)

https://github.com/ppy/osu/assets/93432929/0fb6b361-0d12-4835-ab05-df2a5cf453b1

https://github.com/ppy/osu/assets/93432929/d1838662-3243-4003-a165-6ff85195a18b

# osu!stable (First is my video, second is Colgate's)

https://github.com/ppy/osu/assets/93432929/0a42517b-a155-4550-92f1-94c2e27490f2

https://github.com/ppy/osu/assets/93432929/82edd2be-08f2-4cf9-b157-b7b0853f21da

### Version

2023.924.0

### Logs

My logs:
[performance.log](https://github.com/ppy/osu/files/12708213/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12708214/runtime.log)
[database.log](https://github.com/ppy/osu/files/12708215/database.log)

Colgate Enjoyer's logs:
[performance-colgate.log](https://github.com/ppy/osu/files/12708202/performance-colgate.log)
[runtime-colgate.log](https://github.com/ppy/osu/files/12708203/runtime-colgate.log)
[database-colgate.log](https://github.com/ppy/osu/files/12708212/database-colgate.log)

**Comment by peppy** [comment, marked as answer]
There's an intentional delay to improve performance (reducing the number of background load and difficulty calculation operations). This was from a time when lazer historically performed very bad under these scenarios and without the delay there would be frame drops.

We'll continue to fine tune this going forward – and probably reduce it to be closer to what you expect coming from stable.

**Reply by Alawapr** [reply]
Welp that sucks... Especially as the song selection still lags/causes lag spikes from time to time...
I have a lot of faith in lazer though, hoping the next focus after ranked play is performance 🙏

**Reply by peppy** [reply]
I've reduced the delay drastically in https://github.com/ppy/osu/pull/24929, so look forward to an improvement in the next release.
